tests/catch/unit/memory/hipMalloc3DArray.cc

```diff
@@ -176,20 +126,145 @@ TEST_CASE("Unit_hipMalloc3DArray_MultiThread") {
   std::vector<std::thread> threadlist;
   int devCnt = 0;
   devCnt = HipTest::getDeviceCount();
-  size_t tot, avail, ptot, pavail;
-  HIP_CHECK(hipMemGetInfo(&pavail, &ptot));
+  const auto pavail = getFreeMem();
   for (int i = 0; i < devCnt; i++) {
-    threadlist.push_back(std::thread(Malloc3DArrayThreadFunc, i));
+    // FIXME: HIP_CHECK and HIPASSERT are not threadsafe so this test is broken.
+    threadlist.push_back(std::thread(Malloc3DArray_DiffSizes, i));
   }
 
-  for (auto &t : threadlist) {
+  for (auto& t : threadlist) {
     t.join();
   }
-  HIP_CHECK(hipMemGetInfo(&avail, &tot));
+  const auto avail = getFreeMem();
 
   if (pavail != avail) {
     WARN("Memory leak of hipMalloc3D API in multithreaded scenario");
     REQUIRE(false);
   }
 }
 
+namespace {
+void checkArrayIsExpected(hipArray_t array, const hipChannelFormatDesc& expected_desc,
```

Comment:
What's the point of this function?

Reply:
like #2713, I have updated it do actually do something for the NVIDIA backend.